tests/sentry/api/endpoints/test_sentry_apps.py

```diff
@@ -353,6 +356,106 @@ def test_same_name_internal_integration(self):
         assert response.data["name"] == sentry_app.name
         assert response.data["slug"] != sentry_app.slug
 
+    def test_cannot_create_app_without_organization(self):
+        self.create_project(organization=self.org)
+        self.login_as(user=self.user)
+        sentry_app = self.create_internal_integration(name="Foo Bar")
+        body = {
+            "name": sentry_app.name,
+            "organization": None,
+        }
+        response = self._post(**body)
+        assert response.status_code == 400
+        assert response.data == {
+            "organization": "Please provide a valid value for the 'organization' field.",
+        }
+
+    def test_cannot_create_app_in_alien_organization(self):
+        self.create_project(organization=self.super_org)
+        self.login_as(user=self.user)
+        sentry_app = self.create_internal_integration(name="Foo Bar")
+        body = {
+            "name": sentry_app.name,
+            "organization": self.super_org.slug,
+        }
+        response = self._post(**body)
+        assert response.status_code == 403
+        assert response.data["detail"].startswith("User does not belong to")
+
+    def test_user_cannot_create_app_in_nonexistent_organization(self):
+        self.create_project(organization=self.org)
+        self.login_as(user=self.user)
+        sentry_app = self.create_internal_integration(name="Foo Bar")
+        body = {
+            "name": sentry_app.name,
+            "organization": "some-non-existent-org",
+        }
+        response = self._post(**body)
+        assert response.status_code == 403
+        assert response.data["detail"].startswith("User does not belong to")
+
+    def test_superuser_cannot_create_app_in_nonexistent_organization(self):
+        self.create_project(organization=self.super_org)
+        self.login_as(user=self.superuser, superuser=True)
+        sentry_app = self.create_internal_integration(name="Foo Bar")
+        body = {
+            "name": sentry_app.name,
+            "organization": "some-non-existent-org",
+        }
+        response = self._post(**body)
+        assert response.status_code == 400
+        assert response.data == {
+            "organization": "Organization 'some-non-existent-org' does not exist.",
+        }
+
+    def test_internal_sentry_app_cannot_create_app(self):
+        self.create_project(organization=self.internal_org)
+        sentry_app = self.internal_app
+        body = {
+            "name": sentry_app.name,
+            "organization": self.internal_org.slug,
+        }
+        token = ApiToken.objects.get(application=sentry_app.application)
+        response = self._post_with_token(token, **body)
+        assert response.status_code == 403
+        assert response.data["detail"].startswith("You do not have permission")
+
+    def test_internal_sentry_app_cannot_create_app_without_organization(self):
```

Comment:
@jverce I don't think we really need this test or the two below it. The `test_internal_sentry_app_cannot_create_app` test is checking the spec sufficiently.